tests/acceptance/test_update_control.py

```diff
@@ -0,0 +1,431 @@
+                "fail_after": BETWEEN_EXPIRATIONS,
+                "success": True,
+                "last_successful_state": "ArtifactCommit",
+            },
+        },
+        {
+            "name": "Continue, then fail in ArtifactInstall_Enter",
+            "case": {
+                "maps": [
+                    {
+                        "id": MUID,
+                        "states": {
+                            "ArtifactInstall_Enter": {
+                                "action": "continue",
+                                "on_action_executed": "fail",
+                            },
+                        },
+                    }
+                ],
+                "second_deployment": True,
+                "success": False,
+                "last_successful_state": "Download",
+            },
+        },
+        {
+            "name": "continue with higher priority",
+            "case": {
+                "maps": [
+                    {
+                        "id": MUID,
+                        "states": {"ArtifactInstall_Enter": {"action": "fail",},},
+                    },
+                    {
+                        "id": MUID2,
+                        "states": {"ArtifactInstall_Enter": {"action": "continue",},},
+                        "priority": 1,
+                    },
+                ],
+                # "continue" has lowest precedence, even if the priority is
+                # higher.
+                "success": False,
+                "last_successful_state": "Download",
+            },
+        },
+        {
+            "name": "force_continue with higher priority",
+            "case": {
+                "maps": [
+                    {
+                        "id": MUID,
+                        "states": {"ArtifactInstall_Enter": {"action": "pause",},},
+                    },
+                    {
+                        "id": MUID2,
+                        "states": {
+                            "ArtifactInstall_Enter": {"action": "force_continue",},
+                        },
+                        "priority": 1,
+                    },
+                ],
+                "success": True,
+                "last_successful_state": "ArtifactCommit",
+            },
+        },
+        {
+            "name": "force_continue with lower priority",
+            "case": {
+                "maps": [
+                    {
+                        "id": MUID,
+                        "states": {"ArtifactCommit_Enter": {"action": "fail",},},
+                    },
+                    {
+                        "id": MUID2,
+                        "states": {
+                            "ArtifactCommit_Enter": {"action": "force_continue",},
+                        },
+                        "priority": -1,
+                    },
+                ],
+                "success": False,
+                "last_successful_state": "ArtifactVerifyReboot",
+            },
+        },
+    ]
```

Comment:
Excellent tests. I was able to follow all the cases with no much trouble.

Reply:
That's great! I was afraid the many labels would be a bit too much. At one point did I wonder if it should be split, but there is a lot of reuse.